src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,176 @@
+        ];
+    }
+
+    public function __construct(
+        private readonly Ncr        $ncr,
+        private readonly Key        $key,
+        private readonly Flags      $flags,
+        private readonly LoggerInterface $logger,
+    )
+    {
+    }
+
+    public function handleCommand(Message $command, Pbjx $pbjx): void
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_disabled')) {
+            return;
+        }
+
+        $searchEngines = $command->get('search_engines', ['google']);
+        $nodeRef = $command->get('node_ref');
+
+        try {
+            $node = $this->ncr->getNode($nodeRef);
+        } catch (NodeNotFound $e) {
+            $this->logger->error('Unable to get node for search engines processing.', [
+                'exception' => $e,
+                'node_ref' => $nodeRef
+            ]);
+            return;
+        }
+
+        $retryCommand = clone $command;
+        $retryCommand->clear('search_engines');
+
+        foreach ($searchEngines as $searchEngine) {
+            $methodName = 'checkIndexStatusFor' . ucfirst($searchEngine);
+
+            if (!method_exists($this, $methodName)) {
+                continue;
+            }
+
+            $retryCommand = $this->$methodName($retryCommand, $pbjx, $node);
+        }
+
+        if (empty($retryCommand->get('search_engines'))) {
+            return;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+
+        if ($retries < $maxRetries) {
+            $retryCommand->set('ctx_retries', $retries + 1);
+
+            $pbjx->sendAt(
+                $retryCommand,
+                strtotime($this->flags->getString('inspect_seo_retry_delay', '+15 minutes')),
+                "$nodeRef.inspect-seo"
+            );
+        }
+    }
+
+    public function checkIndexStatusForGoogle(Message $command, Pbjx $pbjx, Message $node): Message
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_google_disabled')) {
+            return $command;
+        }
+
+        $request = new \Google_Service_SearchConsole_InspectUrlIndexRequest();
+        $siteUrl = $this->flags->getString('inspect_seo_google_site_url');
+
+        if (empty($siteUrl)) {
+            $this->logger->error('Must provide a site url.', [
+                'site_url' => $siteUrl,
+            ]);
+
+            return $command;
+        }
+
+        $request->setSiteUrl($siteUrl);
+        $url = UriTemplateService::expand(
+            "{$node::schema()->getQName()}.canonical", $node->getUriTemplateVars()
+        );
+        $request->setInspectionUrl($url);
+        $client = new \Google_Client();
+        $client->addScope(\Google_Service_SearchConsole::WEBMASTERS_READONLY);
+
+        $response = null;
+
+        try {
+            $base64EncodedAuthConfig = Crypto::decrypt(getenv('GOOGLE_SEARCH_CONSOLE_API_SERVICE_ACCOUNT_OAUTH_CONFIG'), $this->key);
```

Comment:
@efjacobson @digitalartcrew in the other uses of google auth variables we don't encrypt them an store them.  I would do like we did with the collect article stats where the config is passed in the constructor and wired in via the symfony yaml for that service. (the ga_secrets_file config)

Reply:
@gdbrown works for me

@digitalartcrew this will turn into something like `$client->setAuthConfig($this->config['google_auth_config']);`